src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorCustomMessageTarget.cs

```diff
@@ -117,6 +127,8 @@ internal partial class RazorCustomMessageTarget
         _telemetryReporter = telemetryReporter;
         _languageServerFeatureOptions = languageServerFeatureOptions;
         _projectSnapshotManagerAccessor = projectSnapshotManagerAccessor;
+        _snippetCache = snippetCache;
+        _xmlSnippetParser = xmlSnippetParser ?? new XmlSnippetParser(null);
```

Comment:
I'm not sure why the [Export] didn't work here, but as long as we're always using the same cache it's okay. Also unsure why we don't have `ILoggerFactory` here but I'm guessing that's just client vs server stuff.

Reply:
I have just enough understanding of our logging to make things appear in logs when I want them, and to recognize that its all weird and probably wrong, but sadly not enough to know how to fix the wrongness :)

I _think_ we would ideally have an `ILoggerFactory` here, and each method in this class would construct its own logger, since they are each endpoints. I'm sure there is an issue logged for better logging somewhere